src/server.py

```diff
@@ -16,13 +18,65 @@ def server():
         conn, addr = server.accept()
         buffer_length = 32
         message_complete = False
-        part = b''
+        full_mes = b''
         while not message_complete:
             partial_mes = conn.recv(buffer_length)
-            part += partial_mes
+            full_mes += partial_mes
             if len(partial_mes) < buffer_length:
                 message_complete = True
-        print(u'received message: ', part.decode('utf8'))
-        conn.sendall(part)
+                full_mes_decoded_to_unicode = full_mes.decode('utf8')
+        print(u'request:\r\n', full_mes_decoded_to_unicode)
+        conn.sendall(response_ok())
         conn.close()
         server.close()
+
+
+def response_ok():
+    """Generate response_ok."""
+    lines = []
+    first_line = 'HTTP/1.1 200 OK'
+    lines.append(first_line)
+    headers = {}
+    body = 'Success!'
+    headers['Content-Type'] = 'text/plain; charset=utf-8'
+    headers['Content-Length'] = str(len(body.encode('utf8')))
+    now = email.utils.formatdate(usegmt=True)
+    headers['Date'] = now
+    header_lines = [': '.join(item) for item in headers.items()]
+    lines.extend(header_lines)
+    lines.append('')
+    lines.append(body)
+    CRLF = '\r\n'
+    response = CRLF.join(lines)
+    response = response.encode('utf8')
+    return response
+
+
+def response_deconstructor(response):
+    """Deconstruct response into basic components.
+    Return protocol, status, status message,
+    headers as dictionary, length of body, number of components
+    separated by first empty line."""
+    response_msg = response
+    uresponse = response_msg.decode('utf8')
+    CRLF = '\r\n'
+    uresponse.split(CRLF)
+    first_pass = uresponse.split(CRLF+CRLF, 1)
+    head, body = first_pass
+    head_lines = head.split(CRLF)
+    protocol, status, msg = head_lines[0].split()
+    headers = head_lines[1:]
+    headers_split = [header.split(':', 1) for header in headers]
+    headers_dict = {k.lower(): v.strip() for k, v in headers_split}
+    return [protocol, status, msg, headers_dict,
+            str(len(body)), str(len(first_pass))]
```

Comment:
Most of this I get, but why the len of `first_pass` at the end? How do you plan on using that?